android/libraries/rib-router-navigator/src/main/kotlin/com/uber/rib/core/StackRouterNavigator.kt

```diff
@@ -198,13 +211,18 @@ open class StackRouterNavigator<StateT : RouterNavigatorState>(private val hostR
     navigationStack.clear()
   }
 
-  private fun buildNewState(
+  @Suppress("UNCHECKED_CAST")
+  private fun <R : Router<*>> buildNewState(
```

Comment:
Why UNCHECKED_CAST? Try to use covariance instead

Reply:
Tried to do so before, but with covariance we loose router type needed in  the `attachInternal` and `detachInternal` methods. I working on code change to avoid this casts, will update PR soon.